iothub/device/src/Transport/Amqp/AmqpConnectionPool.cs

```diff
@@ -9,17 +9,12 @@
 
 namespace Microsoft.Azure.Devices.Client.Transport.Amqp
 {
-    internal class AmqpConnectionPool : IAmqpUnitManager
+    internal sealed class AmqpConnectionPool : IAmqpUnitManager
     {
         private AmqpConnectionHolder[] _amqpSasIndividualPool;
         private readonly Dictionary<string, AmqpConnectionHolder[]> _amqpSasGroupedPool = new();
         private readonly object _lock = new();
 
-        protected virtual IDictionary<string, AmqpConnectionHolder[]> GetAmqpSasGroupedPoolDictionary()
```

Comment:
If not protected, don't need this method, but even if we did, a property with private setter and protected getter would be simpler.